addons/purchase_stock/tests/test_reordering_rule.py

```diff
@@ -950,12 +950,12 @@ def test_decrease_qty_multi_step_receipt(self):
             with form.order_line.edit(0) as line:
                 line.product_qty = 4
         purchase.button_confirm()
-        moves = self.env['stock.move'].search([('id', '!=', out_move.id), ('product_id', '=', self.product_01.id)], order='id desc')
+        moves = self.env['stock.move'].search([('id', '!=', out_move.id), ('name', '=', self.product_01.name)], order='id desc')
```

Comment:
Filtering by name is an atrocity. Let's rely on an accurate criterion to filter out inventory adjustments instead:
```suggestion
        moves = self.env['stock.move'].search([('id', '!=', out_move.id), ('product_id', '=', self.product_01.id), ('is_inventory', '=', False)], order='id desc')
```
The same comment holds for the 2 other changes.